Source/com/drew/metadata/mp4/boxes/ItemListBox.java

```diff
@@ -0,0 +1,145 @@
+import java.util.Map;
+
+
+public class ItemListBox extends Box
+{
+    private class ItemListEntry
+    {
+        private int tag;
+        private String val;
+
+        ItemListEntry(int tag) {
+            this.tag = tag;
+            this.val = null;
+        }
+
+        int getTag() {
+            return this.tag;
+        }
+
+        void setVal(String val) {
+            this.val = val;
+        }
+
+        String getVal() {
+            return this.val;
+        }
+    }
+
+    public static Map<String, ItemListEntry> itemMap;
+
+    {
+        itemMap = new HashMap<String, ItemListEntry>();
+
+        itemMap.put("�alb", new ItemListEntry(Mp4MetaBoxDirectory.TAG_ALBUM));
+        itemMap.put("aART", new ItemListEntry(Mp4MetaBoxDirectory.TAG_ALBUM_ARTIST));
+    }
+
+    public ItemListBox(SequentialReader reader, Box box) throws IOException
+    {
+        super(box);
+
+        long totalRead = 0;
+        try {
+            while (totalRead < box.size) {
+                long recordLen = reader.getUInt32();
+                String fieldName = reader.getString(4);
+                long fieldLen = reader.getUInt32();
+                String typeName = reader.getString(4);//data
+                totalRead += 16;
+                if ("data".equals(typeName)) {
+                    reader.skip(8);//not sure what these are
+                    totalRead += 8;
+                    int toRead = (int) fieldLen - 16;
+                    if (toRead <= 0) {
+                        //log?
+                        return;
+                    }
+                    if ("covr".equals(fieldName)) {
+                        //covr can be an image file, e.g. png or jpeg
+                        //skip this for now
+                        reader.skip(toRead);
+                    } else if ("cpil".equals(fieldName)) {
+                        int compilationId = (int) reader.getByte();
+//                    metadata.set(XMPDM.COMPILATION, compilationId);
+                    } else if ("trkn".equals(fieldName)) {
+                        if (toRead == 8) {
+                            long numA = reader.getUInt32();
+                            long numB = reader.getUInt32();
+//                        metadata.set(XMPDM.TRACK_NUMBER, (int)numA);
+                        } else {
+                            //log
+                            reader.skip(toRead);
+                        }
+                    } else if ("disk".equals(fieldName)) {
+                        int a = reader.getInt32();
+                        short b = reader.getInt16();
+//                    metadata.set(XMPDM.DISC_NUMBER, a);
+                    } else {
+                        String val = reader.getString(toRead);
+                        if (itemMap.containsKey(fieldName)) {
+                            ItemListEntry entry = itemMap.get(fieldName);
+                            entry.setVal(val);
+                            itemMap.put(fieldName, entry);
```

Comment:
I didn't really touch the "ilst" processing part except for right here. We add everything to the item map if it is known.